src/transformStateWithClones.js

```
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  const copyOfState = JSON.parse(JSON.stringify(state));
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const item in action.extraData) {
+          copyOfState[item] = action.extraData[item];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const item of action.keysToRemove) {
+          delete copyOfState[item];
+        }
+        break;
+
+      case 'clear':
+        for (const item in copyOfState) {
+          delete copyOfState[item];
+        }
+        break;
+
+      default:
+        return '';
```

let's throw an error here (if we do not support a certain type of action)